### ISSUE TITLE
Consider if auth tokens should be removed from channel URLs in conda-meta JSON files

### ISSUE THREAD

**Issue by ijstokes** [conda/conda#5955]
Assuming conda is aware of URL-embedded auth tokens (see #5750 and https://github.com/Anaconda-Platform/anaconda-server/issues/1304) it seems like it might be problematic to embed them into the `conda-meta/package-version.json` files for a few reasons:

1. the package cache may be a shared resource and Bob shouldn't have access to Alice's auth tokens.

2. the auth token may expire or be denied and future operations that need to know "where to get the package" should provide a set of auth tokens relevant at that point in time (which may be the same ones), not assume that auth tokens used in the past are going to be OK.

It appears that `conda create --clone` uses this information (though for reasons stated in #5953 perhaps it shouldn't!)

Here are examples from files in my package cache:

`anaconda/conda-meta/history`
```
==> 2017-05-17 08:41:04 <==
# cmd: /Users/ijstokes/anaconda/bin/conda install --yes --json --force-pscheck --prefix /Users/ijstokes/anaconda --override-channels --channel https://repo.continuum.io/pkgs/pro --channel https://repo.continuum.io/pkgs/free --channel https://repo.continuum.io/pkgs/r --channel https://conda.anaconda.org/t/ij-3e473a13-904f-433a-b6ad-SOMESTUFF/anaconda-fusion --channel https://conda.anaconda.org/t/ij-3e473a13-904f-433a-b6ad-SOMESTUFF/conda-forge keras tensorflow theano
+conda-forge::keras-2.0.2-py36_1
+conda-forge::protobuf-3.3.0-py36_0
+conda-forge::tensorflow-1.0.0-py36_0
+defaults::libgpuarray-0.6.4-0
+defaults::mako-1.0.6-py36_0
+defaults::mock-2.0.0-py36_0
+defaults::pbr-1.10.0-py36_0
+defaults::pygpu-0.6.4-py36_0
+defaults::theano-0.9.0-py36_0
# install specs: keras,tensorflow,theano
```

`/Users/ijstokes/anaconda/conda-meta/keras-2.0.2-py36_1.json`
```
"channel": "https://conda.anaconda.org/t/ij-3e473a13-904f-433a-b6ad-SOMESTUFF/conda-forge/osx-64",
"url": "https://conda.anaconda.org/t/ij-3e473a13-904f-433a-b6ad-SOMESTUFF/conda-forge/osx-64/keras-2.0.2-py36_1.tar.bz2",
```

`/Users/ijstokes/anaconda/conda-meta/tensorflow-1.0.0-py36_0.json`
```
"channel": "https://conda.anaconda.org/t/ij-3e473a13-904f-433a-b6ad-SOMESTUFF/conda-forge/osx-64",
"url": "https://conda.anaconda.org/t/ij-3e473a13-904f-433a-b6ad-SOMESTUFF/conda-forge/osx-64/tensorflow-1.0.0-py36_0.tar.bz2",
```

**Comment by kalefranz** [comment]
I've been wrestling with the right way to do this.  I still haven't arrived at an answer I'm satisfied with.  It's definitely on my mind though, and as far as I'm concerned, still not resolved for conda 4.4.

**Comment by kalefranz** [comment]
We're no longer recording the token in the "channel" field.
